primitives/src/signatures/schnorr.rs

```diff
@@ -299,7 +300,7 @@ where
         let mut msg_input = vec![instance_description, fr_to_fq::<F, P>(&self.sk.0)];
         msg_input.extend(msg.iter());
 
-        let r = fq_to_fr::<F, P>(&RescueCRHF::sponge_with_padding(&msg_input, 1)[0]);
+        let r = fq_to_fr::<F, P>(&VariableLengthRescueCRHF::<F, 1>::evaluate_safe(&msg_input)[0]);
```

Comment:
Now that I see this coded up, I prefer removing `evaluate_safe` function, and instead simply have the caller `unwrap` . The behaviour will be exactly the same - but the interface will be cleaner. What do others think?

Reply:
I can see where you are coming from. it's more explicit about the function contract (what it promises) instead of some hidden, less obvious choice of potential panicking.

I will change to `.unwarp()` as suggested